README.md

```diff
@@ -1,2 +1,117 @@
+3. `curl` is installed and available on the PATH: https://everything.curl.dev/get
+
+### Deploy and Setup
+
+Deploy SPIRE server and SPIRE agent in the `spire` namespace on the cluster.
+The script also creates necessary registrations in SPIRE server.
+```
+./01-deploy-spire.sh
+```
+
+Deploy MySQL server along with a persistent volume in the `mysql` namespace on the cluster
+```
+./02-deploy-mysql.sh
+```
+
+In a separate terminal, start port forwarding to MySQL server pod
+```
+kubectl -n mysql port-forward pod/mysql-0 3306:3306
+```
+
+Setup MySQL users and database, by connecting to the MySQL server via port-forwarding.
+```
+./03-setup-mysql.sh
+```
+
+Deploy `sample-service` in the default namespace.
+```
+./04-deploy-service.sh
+```
+
+### Verify DB Access using Sample Service
+
+This section verifies DB access via mTLS using Sample Service.
+
+In a separate terminal, start port forwarding to the `sample-service` 
+```
+kubectl port-forward service/sample-service  8888:8888
+```
+
+Use `curl` to verify the `GET /api/v1/users` endpoint on `localhost:8888`. The output should contain sample user data which was created
+during MySQL DB setup.
+```
+curl -s -X GET http://localhost:8888/api/v1/users
+```
+
+Use `curl` to create a new user 
+```
+curl -s -X POST http://localhost:8888/api/v1/users -H 'Content-Type: application/json' -d '{"Name":"David"}'
+```
+
+Verify the newly created user is showing up in the `GET /api/v1/users` request
+```
+curl -s -X GET http://localhost:8888/api/v1/users
+```
+
+Verify the DB connection made by `sample-service` in the MySQL server general log. It should have a `Connect` log message from 
+`spire-mysql-client` user (representing sample-service) connecting to the `spiredemo` DB using SSL/TLS.
+```
+kubectl exec -it mysql-0 -n mysql -c mysql -- cat /var/lib/mysql/general.log
+```
+
+### Cleanup 
+
+Cleanup the environment using the cleanup script
+```
+./cleanup.sh
+```
+
+## [Design](#design)
+
+### Architecture
+
+![Architecture](./docs/img/architecture.png)
+
+MySQL server uses its X.509-SVIDs as server certificate and encryption whereas clients use the X.509-SVIDs to
+authenticate to the MySQL server. SPIRE Server and SPIRE Agent are deployed on the Kubernetes Cluster along
+with registration entries created in the  SPIRE Server. MySQL is deployed with an init-container that fetches
+X.509-SVID from SPIRE agent and writes to a tmpfs volume shared with the MySQL server container.
+Upon init, MySQL server container reads the X.509-SVID from the tmpfs volume for SSL configuration.
+
+The sample-service also fetches its X.509 SVID from SPIRE and uses it to authenticate to MySQL server and  
+to verify MySQL server's certificate (mTLS). The design also uses SPIRE CredentialComposer plugin to configure
+X.509-SVIDs issued to sample-service have a unique Subject with Common Name - `/C=US/O=SPIRE/CN=spire-mysql-client`,
+thereby mapping this service's identity to the user created in MySQL -
```

Comment:
We should also note that this is needed because MySQL doesn't know how to do SPIFFE authentication using verification of SPIFFE IDs in the URI SAN, i.e. default X.509-SVIDs issued by SPIRE don't work for this case